crypto/dsa/dsa_test.go

```diff
@@ -0,0 +1,181 @@
+package dsa_test
+
+import (
+	"testing"
+
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+	"github.com/tbd54566975/web5-go/crypto/dsa/ecdsa"
+	"github.com/tbd54566975/web5-go/crypto/dsa/eddsa"
+)
+
+func TestGeneratePrivateKeySECP256K1(t *testing.T) {
+	privateJwk, err := dsa.GeneratePrivateKey(dsa.AlgorithmID.SECP256K1)
+	if err != nil {
+		t.Errorf("failed to generate private key: %v", err.Error())
+	}
+
+	if privateJwk.CRV != ecdsa.SECP256K1JWACurve {
+		t.Errorf("unexpected CRV. expected: %v got: %v", ecdsa.SECP256K1JWACurve, privateJwk.CRV)
+	}
+
+	if privateJwk.KTY != ecdsa.KeyType {
+		t.Errorf("unexpected KTY. expected: %v got: %v", ecdsa.KeyType, privateJwk.KTY)
+	}
+
+	if privateJwk.D == "" {
+		t.Errorf("privateJwk.D is empty")
+	}
+
+	if privateJwk.X == "" || privateJwk.Y == "" {
+		t.Errorf("privateJwk.X or privateJwk.Y is empty")
+	}
+}
```

Comment:
I would suggest just using an assertion library. There are fanatics in the Go community that prefer to use just the standard library, but an assertion library removes a _lot_ of noise. eg.

```go
	privateJwk, err := dsa.GeneratePrivateKey(dsa.AlgorithmID.SECP256K1)
    assert.NoError(t, err)
    assert.Equal(t, privateJwk.CRV, ecdsa.SECP256K1JWACurve)
    assert.Equal(t, privateJwk.KTY, ecdsa.KeyType)
    assert.NotZero(t, privateJwk.D)
    assert.True(t, privateJwk.X != "" && privateJwk.Y != "", "privateJwk.X or privateJwk.Y is empty")
```

vs.

```go
	if err != nil {
		t.Errorf("failed to generate private key: %v", err.Error())
	}

	if privateJwk.CRV != ecdsa.SECP256K1JWACurve {
		t.Errorf("unexpected CRV. expected: %v got: %v", ecdsa.SECP256K1JWACurve, privateJwk.CRV)
	}

	if privateJwk.KTY != ecdsa.KeyType {
		t.Errorf("unexpected KTY. expected: %v got: %v", ecdsa.KeyType, privateJwk.KTY)
	}

	if privateJwk.D == "" {
		t.Errorf("privateJwk.D is empty")
	}

	if privateJwk.X == "" || privateJwk.Y == "" {
		t.Errorf("privateJwk.X or privateJwk.Y is empty")
	}
```

I'm biased, but I'd suggest using https://github.com/alecthomas/assert as it's very small, and type safe.

Reply:
oh wait, I misunderstood... @alecthomas you're referring specifically to tests for using assertion libraries? Or both test and actual sdk source code?

Reply:
ah yeah @alecthomas i remember you mentioning your assert library a few weeks ago. totally agree. figured i'd create a separate issue to make the switch when i thought this PR was going to be small 🙈 . given that it's the polar opposite of that, will switch over now.

Reply:
For tests only, yeah. It's not uncommon to use `panic()` in Go, in similar situations to where you'd use `assert` in C though.